code/datums/viewport.dm

```diff
@@ -0,0 +1,72 @@
+/client/ // Holder Variables
+	var/last_view_x_dim = 7
+	var/last_view_y_dim = 7
+
+// Used in OnResize() - Holds the clients prefs.
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
+
+// We hold this so we don't hit prefs often.
+/datum/preferences/var/icon_size = 64
```

Comment:
Do single vars like this
also why is this here
```suggestion
/datum/preferences
	var/icon_size = 64
```

Reply:
Where would it properly go?

Reply:
see below, I think you should use the preferences ui/backend for this

Reply:
done